### PR TITLE
Global layouts error boundary

### PR DESCRIPTION
**Pull request by huozhi**
Adding global error boundary for appDir, when there's an error occurred and no error component there as fallback, will use the global error fallback

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [x] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see `contributing.md`

### CODE REVIEW

File: packages/next/client/components/app-router.client.tsx [hunk 1]
```diff
@@ -353,13 +354,15 @@ export default function AppRouter({
...
               }}
             >
-              {HotReloader ? (
-                <HotReloader assetPrefix={assetPrefix}>
-                  {cache.subTreeData}
-                </HotReloader>
-              ) : (
-                cache.subTreeData
-              )}
+              <ErrorBoundary errorComponent={GlobalErrorComponent}>
```
Comment (timneutkens) [inline packages/next/client/components/app-router.client.tsx:357]: This boundary should wrap around the AppRouter component instead.

Changing the default export to something like:

```
export default function AppRouter(props) {
  return <ErrorBoundary errorComponent={GlobalErrorComponent}>
     <Router {...props} />
  </ErrorBoundary>
}
```

The existing function would be renamed to `Router` in this.

File: test/e2e/app-dir/index.test.ts [hunk 2]
```diff
@@ -1494,57 +1501,89 @@ describe('app dir', () => {
...
-            .waitForElementByCss('#error-trigger-button')
-
-          expect(
-            await browser.elementByCss('#error-trigger-button').text()
-          ).toBe('Trigger Error!')
         }
       })
 
-      it('should hydrate empty shell to handle server-side rendering errors', async () => {
+      it('should fallback default error boundary when no error component specified', async () => {
```
Comment (timneutkens) [inline test/e2e/app-dir/index.test.ts:1524]: The fallback error boundary should never show in development as the error overlay boundary should have caught it 🤔

Comment (huozhi) [reply test/e2e/app-dir/index.test.ts:1524]: Yes the tests fallback to the global one only in prod, use the dev overlay in dev mode. need to rewrite the test case name

Comment (huozhi) [inline test/e2e/app-dir/index.test.ts:1524]:
```suggestion
      it('should use default error boundary for prod and overlay for dev when no error component specified', async () => {
```

**Review by timneutkens — APPROVED** [review]

**Pull request merged**